corpus/function_declarations.txt

```diff
@@ -399,6 +399,48 @@ f a = c
         (variable_identifier)
         (function_body (integer)))))))
 
+==============================================================
+Function Declarations With Let and In on last declaration-line
+==============================================================
+
+f = let y = x
+        z = 2
+        x = 1 in y
+
+f = let y = 42 in y
+
+---
+
+(module
+  (function_declaration
+    (variable_identifier)
+    (function_body
+      (let_expression
+        (function_declaration
+          (variable_identifier)
+          (function_body
+            (variable_identifier)))
+        (function_declaration
+          (variable_identifier)
+          (function_body
+            (integer)))
+        (function_declaration
+          (variable_identifier)
+          (function_body
+            (integer)))
+        (in_clause
+          (variable_identifier)))))
+  (function_declaration
+    (variable_identifier)
+    (function_body
+      (let_expression
+        (function_declaration
+          (variable_identifier)
+          (function_body
+            (integer)))
+        (in_clause
+          (variable_identifier))))))
+
```

Comment:
@clojj I'm not sure I see how these test examples improve on the existing [let tests](https://github.com/tree-sitter/tree-sitter-haskell/blob/568d75da23c392208351b55594da643526e96ef8/corpus/function_declarations.txt#L402-L419). Perhaps I'm missing something?

Reply:
I will probably remove these tests... will look into it

Reply:
> One other thing - this parser will not work correctly in Atom quite yet. The external scanner uses a new Tree-sitter feature (the get_column method on TSLexer) which incremental parsing support is not fully implemented. Subscribe to tree-sitter/tree-sitter#144 for updates on this.

So do I understand correctly that get_column() is actually implemented but not yet cooperating with the incremental stuff... but otherwise returns correct (column-)values ?

Reply:
Exactly.